Sources/AsyncHTTPClient/ConnectionsState.swift

```diff
@@ -285,22 +259,14 @@ extension HTTP1ConnectionProvider {
             }
         }
 
-        mutating func processNextWaiter() -> Action {
+        mutating func processNextWaiter() -> Action<ConnectionType> {
             if let waiter = self.waiters.popFirst() {
-                let (eventLoop, required) = self.resolvePreference(waiter.preference)
+                // There should be no case where we have waiters and available connections. As soon as connection
+                // is released and can be available, it should immideately go to the next waiter.
+                assert(self.availableConnections.isEmpty)
 
-                // If specific EL is required, we have only two options - find open one or create a new one
-                if required, let found = self.availableConnections.firstIndex(where: { $0.channel.eventLoop === eventLoop }) {
```

Comment:
the following block was remove since it cannot be reached, we replaced it with an assert on line 266